nncf/tensorflow/quantization/initializers/collectors.py

```diff
@@ -11,268 +11,129 @@
  limitations under the License.
 """
 
-from typing import Tuple, List, Union
+from collections import deque
+from typing import List
 
 import numpy as np
 import tensorflow as tf
 
-from nncf.tensorflow.layers.operation import InputType
+from nncf.common.tensor_statistics.collectors import OfflineTensorStatisticCollector
+from nncf.common.tensor_statistics.collectors import MixedMinMaxStatisticCollector
+from nncf.common.tensor_statistics.collectors import MeanMinMaxStatisticCollector
+from nncf.common.tensor_statistics.collectors import MinMaxStatisticCollector
+from nncf.common.tensor_statistics.collectors import ReductionShape
+from nncf.common.tensor_statistics.statistics import MedianMADTensorStatistic
+from nncf.common.tensor_statistics.statistics import PercentileTensorStatistic
 from nncf.tensorflow.quantization.initializers.utils import get_per_channel_history
 from nncf.tensorflow.quantization.initializers.utils import discard_zeros
-from nncf.tensorflow.quantization.initializers.utils import get_axes
+from nncf.tensorflow.tensor import TFNNCFTensor
 
 
-class MinMaxStatisticCollector:
-    """
-    Collector uses min of minimum values and max of maximum values.
-    """
-
-    def __init__(self, per_channel: bool, channel_axes: Union[int, Tuple[int], List[int]], input_type: str):
-        self._per_channel = per_channel
-        self._channel_axes = channel_axes if isinstance(channel_axes, (list, tuple)) else [channel_axes]
-        self._input_type = input_type
-        self._all_min_values = []
-        self._all_max_values = []
-
-    @property
-    def min(self) -> tf.Tensor:
-        return tf.math.reduce_min(tf.stack(self._all_min_values), axis=0)
-
-    @property
-    def max(self) -> tf.Tensor:
-        return tf.math.reduce_max(tf.stack(self._all_max_values), axis=0)
+class TFMinMaxStatisticCollector(MinMaxStatisticCollector):
+    def _register_input(self, x: tf.Tensor):
```

Comment:
let's discuss and choose a type for tf tensor. Can be or `tf.Tensor` or `tf.Variable`
```suggestion
    def _register_input(self, x: tf.Varible):
```

Reply:
Discussed and decided to use the tf.Tensor